Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3148,6 +3152,15 @@ private unsafe void OnUpdate(InputUpdateType updateType, ref InputEventBuffer ev
                     }
                 }
 
+                // Give the device a chance to do something with data before we propagate it to event listeners.
+                if (!skipEvent && device.hasEventPreProcessor)
```

Comment:
Should we add some tests for preprocess logic?

Reply:
+100

Reply:
fixed